DESIGN.md

```diff
@@ -0,0 +1,98 @@
+# Bulk Data Extract (BDE) Processor design
+
+## User requirements
```

Comment:
This repo is already open, and I'm planning to keep all the docs here rather than in internal systems. That said, what form should the information about the layers take? Presumably users have no knowledge of the BDE format or the Kx internal APIs.